pkg/chunk/series_store.go

```diff
@@ -298,15 +295,25 @@ func (c *seriesStore) lookupSeriesByMetricNameMatchers(ctx context.Context, from
 			incomingIDs <- ids
 		}(matcher)
 	}
-
+	// Just get series for metric if there are no matchers after exclusions
+	if len(matchers) == 0 || counter == 0 {
```

Comment:
I don't think the first part can be true without the second also being true, i.e. the first is redundant.

Reply:
Done!